src/dsp_tools/utils/shared.py

```diff
@@ -24,46 +24,43 @@
 
 def login(server: str, user: str, password: str) -> Connection:
     """
-    Logs in and returns the active connection.
+    Creates a connection, 
+    makes a login (handling temporary network interruptions),
+    and returns the active connection.
 
     Args:
         server: URL of the DSP server to connect to
         user: Username (e-mail)
         password: Password of the user
 
     Raises:
-        BaseError if the login fails
+        UserError if the login fails permanently
```

Comment:
what do you mean by "fails permanently"? I guess you want to stress that it failed despite multiple retries. But I feel just "fails" is clear enough